### PR TITLE
add duplicate button

### LINKED ISSUE #1221: Add `duplicate` button
## 💬 Brief description of feature

A Game Designer needs to repeat a number of messages from a past turn.  These will be COA messages, and just need new dates (in the current turn).

We'll temporarily provide a modified version of Serge.  Once the duplications are done, we'll revert to the original version of Serge - so we know we won't have introduced any fragile new features.

## 📖 User Story

As a Game Designer I wish to be able to duplicate a message to save re-typing it.

In a Collab-Working channel, running in collab-edit mode, and if the current user has a release-manager permission, add a `Duplicate` button in the message header (with the other `Action` controls). When this is clicked, create a copy of the current message, strip out the `_id` and `_reference` fields, and add `Copy` to the `Title`, and then send it to the database.

I can do the logic to decide when to display the button, but I'll need @lilitkarapetyan support for the database bits.

**Note:** it's really important that this feature is implemented in a branch of the commit tagged with `20211028` - and not off the current `develop` branch. This is because the Game Designer is going to be editing a Serge database at that version - and we want to be very sure we don't introduce any non-compliant data objects to that game.

## ☑️ Acceptance criteria

- [ ] Tests pass

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
Fixes #1221

## 🔨Work carried out:

- [x] Tests pass

## 🖥️ Screenshot

Screenshot

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

File: packages/components/src/local/molecules/channel-coa-message-detail/helpers/messages_helper.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { COUNTER_MESSAGE } from '@serge/config'
```
Comment (IanMayo) [inline packages/components/src/local/molecules/channel-coa-message-detail/helpers/messages_helper.ts:1]: Hello @lilitkarapetyan 

Once the game managers have duplicated their messages, I expect to throw "Duplicate" button away.

But, the code in this helper function looks like we should keep it.  Can you explain what it is doing please?

**Comment by IanMayo** [comment]
Thanks @lilitkarapetyan 

We're adding this feature to help a wargame that is already running.  We can't add any other new features to that wargame, since they won't have enough testing.  So, we have to add the feature to the exact commit for that version of Serge:

image

If you could switch the PR so it branches off that tag, then I'll step in and look at the logic for when to display the button.

**Comment by IanMayo** [comment]
> Thanks @lilitkarapetyan
> 
> We're adding this feature to help a wargame that is already running. We can't add any other new features to that wargame, since they won't have enough testing. So, we have to add the feature to the exact commit for that version of Serge:
> 
> image
> 
> If you could switch the PR so it branches off that tag, then I'll step in and look at the logic for when to display the button.

@lilitkarapetyan - I'm attempting to add your changes to the legacy target branch.

File: packages/components/src/local/molecules/channel-coa-message-detail/helpers/messages_helper.ts (hunk 1 above)
Comment (lilitkarapetyan) [reply packages/components/src/local/molecules/channel-coa-message-detail/helpers/messages_helper.ts:1]: It's special type of message witch one contain counter for COA messages, it's for creating queue when players updating coa message in some time do to not have conflicts and keep righ counter
https://github.com/serge-web/serge/blob/develop/packages/client/src/api/messages_helper.ts

**Comment by IanMayo** [comment]
Closing. Have moved changes to #1275

**Pull request closed without merging**